<?xml version="1.0"?>
<launch>

  <node pkg="racecar_control" type="gazebo_odometry.py" name="gazebo_odometry_node" output="screen"/>

  <!-- Launch the built-map -->
  <node name="map_server" pkg="map_server" type="map_server" args="$(find racecar_nav)/map/map.yaml" />

  <param name="/use_sim_time" value="true"/>
  <node pkg="move_base" type="move_base" respawn="false" name="move_base" output="screen">
    <rosparam file="$(find jps_global_planner)/config/costmap_common_params.yaml" command="load" ns="global_costmap" />
    <rosparam file="$(find jps_global_planner)/config/costmap_common_params.yaml" command="load" ns="local_costmap" />
    <rosparam file="$(find jps_global_planner)/config/local_costmap_params.yaml" command="load" />
    <rosparam file="$(find jps_global_planner)/config/global_costmap_params.yaml" command="load" />
    <rosparam file="$(find jps_global_planner)/config/teb_local_planner_params.yaml" command="load" />

    <param name="base_global_planner" value="jps_global_planner/JPSGlobalPlanner"/>
    <param name="planner_frequency" value="0.0" />
    <param name="planner_patience" value="20.0" />
    <param name="controller_frequency" value="5.0" />
    <param name="controller_patience" value="20.0" />

    <param name="base_local_planner" value="teb_local_planner/TebLocalPlannerROS" />
    <param name="controller_frequency" value="5.0" />
    <param name="controller_patience" value="15.0" />
    <param name="clearing_rotation_allowed" value="false" />

    <!--<remap from="/cmd_vel" to="/cmd_vel_movebase"/>-->
  </node>

  <include file="$(find racecar_nav)/launch/amcl.launch"/>
  <include file="$(find racecar_nav)/launch/racecar_rviz.launch"/>

  <node pkg="racecar_control" type="cmd_vel_converter.py" name="cmd_vel_converter" output="screen"/>
</launch>

